suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +58,61 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
-            register_sections = xml.findall('register')
-            for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+        baseproduct_filename = get_baseproduct(products_metadata)
+        if len(baseproduct_filename) == 1:
+            baseproduct = baseproduct_filename.pop()
+        else:
+            if not baseproduct_filename:
+                log.error('There is no baseproduct')
+                raise DistMigrationProductSetupException(
+                    'There is no baseproduct'
+                )
+            else:
+                log.error('There are more than 1 product without flavor tag')
```

Comment:
Please update the message, I suggest "Found multiple product definitions without <flavor> element'

Also assign the message to a variable so it doesn't need to be typed twice, once for the exception and once for the log